docs/source/api_reference.rst

```diff
@@ -36,6 +36,30 @@ Utilities to preprocess data before using evalml.
     split_data
 
 
+.. currentmodule:: evalml.exceptions
+
+Exceptions
+=============
+
+.. autosummary::
+    :toctree: generated
+    :template: class.rst
+    :nosignatures:
+
+    MethodPropertyNotFoundError
+    PipelineNotFoundError
+    ObjectiveNotFoundError
+    IllFormattedClassNameError
+    MissingComponentError
```

Comment:
Nice! nit-pick: Is it possible to fix some of the formatting of the docstrings in these exceptions? Ie `all_components()` here:
![image](https://user-images.githubusercontent.com/22552445/104629131-12163e80-5667-11eb-823f-03fb403e5c96.png)